src/types.ts

```diff
@@ -216,3 +245,26 @@ export interface SelectRowEvent {
   checked: boolean;
   isShiftClick: boolean;
 }
+
+export interface Dictionary<T> {
+  [index: string]: T;
+}
+
+export type GroupByDictionary<TRow> = Dictionary<{
+  childRows: TRow[];
+  childGroups: TRow[] | GroupByDictionary<TRow>;
+  startRowIndex: number;
+}>;
+
+export interface GroupRow<TRow> {
+  __isGroup: true;
```

Comment:
We could generate a `Set`/`WeakSet` that contains all the group rows instead of hiding it in plain sight.

```ts
export function isGroupRow<R>(row: unknown, groupRows: ReadonlySet<GroupRow<R>>): row is GroupRow<R> {
  return groupRows.has(row);
}
```

Alternatively we could use a private symbol, though I don't know if that works well with ts.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol
```ts
const rowGroupSymbol = Symbol();

const rowGroup: RowGroup<...> = {
  [rowGroupSymbol]: true
};
```

WDYT?

Reply:
We can use a `Set` that contains all the group rows. This way we won't need `__isGroup `

Reply:
I can try a private symbol

Reply:
We could do the check outside the function?
```
focusable(row: R, isGroupRow: false) {
focusable(row: GroupRow<R>, isGroupRow: true) {
```